### PR TITLE
DAOS-2881 control: Print correct caller info in logging.Debugf()

### PR DESCRIPTION
**Pull request by mjmac**
Use the runtime package to inspect the call stack and adjust
for any Debug() or Debugf() wrappers. Ensures that the correct
caller info is logged.

### CODE REVIEW

**Comment by mjmac** [comment]
Functional failure is our new friend DAOS-4410.

File: src/control/logging/logging_test.go [hunk 1]
```diff
@@ -25,12 +25,59 @@ package logging_test
...
+func (tw *testWrap) Debug(msg string) {
+	tw.log.Debug(msg)
+}
+
+func (tw *testWrap) Debugf(fmtStr string, args ...interface{}) {
+	tw.log.Debugf(fmtStr, args...)
+}
+
+// NB: Keep this test at the top of the file to avoid having to
+// update the line numbers all the time.
```
Comment (kjacque) [inline src/control/logging/logging_test.go]: Not sure if it's feasible, but maybe a simple set of regular expressions could be used to remove this fragility

Comment (mjmac) [reply src/control/logging/logging_test.go]: Well, I wanted to be sure that the reported line number matched the actual line number. Couldn't think of a more robust way to do that. Open to suggestions, though.

Comment (kjacque) [reply src/control/logging/logging_test.go]: Could get the line number with runtime.Caller on the line before or after the call being tested, then add/subtract one, perhaps? I just don't like the idea of this breaking when we have to add a new import or something.

Comment (mjmac) [reply src/control/logging/logging_test.go]: Yeah, good call. I've updated the test to be insulated from line number changes.

File: src/control/logging/debug.go [hunk 2]
```diff
@@ -50,9 +52,24 @@ type DefaultDebugLogger struct {
 // Debugf emits a formatted debug message.
 func (l *DefaultDebugLogger) Debugf(format string, args ...interface{}) {
 	depth := logOutputDepth
-	if len(args) == 0 {
-		// Adjust for the extra call to Debug()
-		depth += 1
+
+	// Adjust depth to account for any convenience wrappers. Enables
+	// printing of correct caller info.
+	pc := make([]uintptr, depth+5)
```
Comment (jbrosenz) [inline src/control/logging/debug.go:58]: What is used to determine that the size is depth+5?  Does +5 need to be configurable?

Comment (jbrosenz) [reply src/control/logging/debug.go:58]: Ok.  Thanks for the explanation.  I am fine with it as-is.  Just wanted to understand it better.

**Comment by mjmac** [comment]
Passed run_tests.sh and small hardware functional testing; no need to waste cluster time on the other functional tests as this commit only affects debug logging output.

**Pull request merged**